infrastructure/local-provisioners/azurerm_apim.ts

```diff
@@ -0,0 +1,510 @@
+  apim_include_policies: t.boolean,
+  apim_task: t.literal("setupOpenapi")
+});
+
+const CreateApiManagementServiceParams = t.interface({
+  environment: t.string,
+  azurerm_apim_eventhub_connstr: t.string,
+  adb2c_tenant_id: t.string,
+  adb2c_portal_client_id: t.string,
+  adb2c_portal_client_secret: t.string,
+  apim_task: t.literal("createApiManagementService")
+});
+
+type ApimParams = t.TypeOf<typeof ApimParams>;
+
+const ApimParams = t.union([
+  SetupOpenApiParams,
+  CreateApiManagementServiceParams
+]);
+
+const addDays = (date: Date, days: number) => {
+  const result = new Date(date);
+  result.setDate(result.getDate() + days);
+  return result;
+};
+
+const setupOpenapi = (
+  apiClient: apiManagementClient,
+  config: IResourcesConfiguration,
+  backendUrl: string,
+  masterKey: string,
+  includeProducts: boolean,
+  includePolicies: boolean
+) => {
+  return Promise.all(
+    config.apim_apis.map(async apiEntry => {
+      const contentValue = `${backendUrl}${apiEntry.api.specsPath}?code=${
+        masterKey
+      }`;
+
+      winston.info(
+        `Adding API from URL: ${backendUrl}${apiEntry.api.specsPath}`
+      );
+
+      // Add API to API management
+      await apiClient.api.createOrUpdate(
+        config.azurerm_resource_group,
+        config.azurerm_apim,
+        apiEntry.id,
+        {
+          contentFormat: "swagger-link-json",
+          contentValue,
+          displayName: apiEntry.api.displayName,
+          path: apiEntry.api.path,
+          protocols: ["https"],
+          // WARNING: serviceUrl is taken from the swagger specs "host" field
+          // and there's no way to override that value here: it *must* be changed
+          // manually in the API management settings
+          // (or provide a real value in the swagger specs).
+          serviceUrl: backendUrl
+        }
+      );
+      // Add API to products
+      if (includeProducts) {
+        await Promise.all(
+          apiEntry.products.map(async (product: string) => {
+            winston.info(
+              `Import API product into the API management: ${product}`
+            );
+            return apiClient.productApi.createOrUpdate(
+              config.azurerm_resource_group,
+              config.azurerm_apim,
+              product,
+              apiEntry.id
+            );
+          })
+        );
+      }
+      // Add a policy to the API reading it from a file
+      if (includePolicies && apiEntry.policyFile) {
+        winston.info(
+          `Import API policy into the API management: ${apiEntry.policyFile}`
+        );
+        const policyContent = fs.readFileSync(
+          path.join(__dirname, "..", "api-policies", apiEntry.policyFile),
+          "utf8"
+        );
+        await apiClient.apiPolicy.createOrUpdate(
+          config.azurerm_resource_group,
+          config.azurerm_apim,
+          apiEntry.id,
+          {
+            policyContent
+          },
+          // If-Match: *
```

Comment:
why is this commented out? not needed?

Reply:
no, it's the description of the following value (which happens to be "*" :)